File: frontend/src/models/Basket.model.ts
import { Dayjs } from 'dayjs';

import { type IPattern } from '@/models/Pattern.model';

export interface IBasket {
    readonly items: Array<IBasketItem>;
    readonly totalPrice: number;
}

export interface IBasketItem {
    readonly pattern: IPattern;
    readonly addedAt: Dayjs;
}

File: backend/ClickStitch/Models/BasketModel.cs
namespace ClickStitch.Models;

public sealed class BasketModel
{
    public required List<BasketItemModel> Items { get; init; }
    public required decimal TotalPrice { get; init; }
}

public sealed class BasketItemModel
{
    public required PatternModel Pattern { get; init; }
    public required DateTime AddedAt { get; init; }
}